### DISCUSSION TITLE
Jest testing Component that is using useRouter()

### DISCUSSION THREAD

**Discussion by vladutcoman** [expo/router#613] [Q&A]
Cheers everyone!

I am trying to test a simple component that is using **useRouter()** and I ca not solve this error: **Couldn't find a navigation object. Is your component inside NavigationContainer?**
I have tried to mock the `useRouter`, but it not helped. Also, since I am using `expo-router`, I do not have `'@react-navigation/native`' installed to get the `NavigationContainer` - correct me if I can take it from anywhere else.

Here is a sample of my code:

```
import { useRouter } from 'expo-router'

const CustomComponent: React.FC<Props> = () => {
  const router = useRouter()

  const onPress = () => {
    router.replace(PATHS.DASHBOARD)
  }

  return <HCard pressAction={onPress} title={'title'} />
}
```
```
describe('CustomComponent', () => {
  it('should render CustomComponent correctly', () => {
    const wrapper = renderWithProvider(<CustomComponent />)
    expect(wrapper.getByText('title')).toBeOnTheScreen()
  })
})
```

```
export const renderWithProvider = (children: JSX.Element): RenderResult => {
  const inset = {
    frame: { x: 0, y: 0, width: 0, height: 0 },
    insets: { top: 0, left: 0, right: 0, bottom: 0 },
  }

  return render(<NativeBaseProvider initialWindowMetrics={inset}>{children}</NativeBaseProvider>)
}
```

**Comment by jar145** [comment]
Why is this closed? I don't see an answer and I am struggling with the same thing
